src/book/03-guides/01-main-codebase/02-tuning-scripts.mdx

```diff
@@ -0,0 +1,137 @@
+
+More on the script system can be found on the [Configuration and Script System page](/system/foundations/config-script).
+
+## Script Tuner
+
+Script tuner is a tool that can be used to create and modify scripts. To use script tuner, run:
+
+```bash
+./scripttuner /path/to/Script.yaml
+```
+
+Where `Script.yaml` is either the name of an existing script, if you want to edit it, or the name for a new script. `/path/to/Script.yaml` is the path to the script. This may be `scripts/nugus/Script.yaml`, or `scripts/nugus*/Script.yaml` where `*` is the robot number for a robot specific script.
+
+Running a command like the above will open the following interface, which will enable you to tune the script:
+
+```
+-----------------------------------Script Tuner-----------------------------------
+
+Script: scripts/nugus2/walk.yaml
+Frames: 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28
+Duration:  3000
+-
+
+
+U Head Pan              Angle:  -.--- Gain: ---.-
+U Head Tilt             Angle:  -.--- Gain: ---.-
+L Right Shoulder Pitch  Angle: +1.963 Gain:  30.0
+L Left Shoulder Pitch   Angle: +1.963 Gain:  30.0
+L Right Shoulder Roll   Angle: -0.124 Gain:  30.0
+L Left Shoulder Roll    Angle: +0.124 Gain:  30.0
+L Right Elbow           Angle: -2.443 Gain:  30.0
+L Left Elbow            Angle: -2.443 Gain:  30.0
+L Right Hip Yaw         Angle: +0.030 Gain:  30.0
+L Left Hip Yaw          Angle: +0.030 Gain:  30.0
+L Right Hip Roll        Angle: -0.063 Gain:  30.0
+L Left Hip Roll         Angle: +0.063 Gain:  30.0
+L Right Hip Pitch       Angle: -0.204 Gain:  30.0
+L Left Hip Pitch        Angle: -0.204 Gain:  30.0
+L Right Knee            Angle: +0.257 Gain:  30.0
+L Left Knee             Angle: +0.257 Gain:  30.0
+L Right Ankle Pitch     Angle: -0.239 Gain:  30.0
+L Left Ankle Pitch      Angle: -0.239 Gain:  30.0
+L Right Ankle Roll      Angle: +0.060 Gain:  30.0
+L Left Ankle Roll       Angle: -0.067 Gain:  30.0
+
+Commands
+, Left a frame  N New Frame        Lock/Unlock   J Jump to Frame P play
+. Right a frame I Delete a Frame T Edit Duration G Change Gains  S Save
+
+Type :help for a full list of commands
+----------------------------------------------------------------------------------
+```
+
+The first line tells you the name and path of the script you are creating or editing.
+
+The second line specifies the number of frames and the current frame, which is highlighted. Each frame specifies different joint (i.e. servo) positions, and when these frames are combined they result in a motion.
+
+The next line is the duration. This determines how long the robot will take to move to the servo positions you have specified. This is measured in milliseconds.
+
+The next section is a table that shows the angle and gain for each servo for that frame.
```

Comment:
We have here another example which implies that `gain` is always 30. If it should always be thirty, could we indicate that here? If it shouldn't, could we change some of the values in the table so it doesn't give that impression?